spark/src/main/java/org/opensearch/sql/spark/response/JobExecutionResponseReader.java

```diff
@@ -46,9 +47,16 @@ private JSONObject searchInSparkIndex(QueryBuilder query, String resultIndex) {
     searchSourceBuilder.query(query);
     searchRequest.source(searchSourceBuilder);
     ActionFuture<SearchResponse> searchResponseActionFuture;
+    JSONObject data = new JSONObject();
     try {
       searchResponseActionFuture = client.search(searchRequest);
     } catch (Exception e) {
+      // if there is no result index (e.g., EMR-S hasn't created the index yet), we return empty
+      // json
+      if (e instanceof IndexNotFoundException) {
```

Comment:
nit: we could catch IndexNotFoundException?

Reply:
Probably build is failing because we are missing unit tests for this piece of code.

Reply:
@vamsi-amazon yeah, adding one now

Reply:
@penghuo good catch. changed.